src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_ComputeModel.cs

```diff
@@ -103,119 +101,44 @@ public void Do()
 
                         // TODO(cyrusn): We're calling into extensions, we need to make ourselves resilient
                         // to the extension crashing.
-                        var groups = await GetGroupsAsync(_completionService, _triggerInfo, cancellationToken).ConfigureAwait(false);
-                        if (groups == null)
+                        var completionList = await GetCompletionListAsync(_completionService, _triggerInfo, cancellationToken).ConfigureAwait(false);
+                        if (completionList == null)
                         {
                             return null;
                         }
 
-                        groups.Do(AddGroupToMap);
-                        if (_displayNameToItemsMap.Count == 0)
-                        {
-                            return null;
-                        }
-
-                        var totalItems = _displayNameToItemsMap.Values.Flatten().ToList();
-                        totalItems.Sort();
-
                         var trackingSpan = await _completionService.GetDefaultTrackingSpanAsync(_documentOpt, _subjectBufferCaretPosition, cancellationToken).ConfigureAwait(false);
                         return Model.CreateModel(
                             _disconnectedBufferGraph,
                             trackingSpan,
-                            totalItems,
-                            selectedItem: totalItems.First(),
+                            completionList.Items.ToList(),
```

Comment:
Hrmm... What's the type of "Items"?  If it's IEnumerable, that's probably not good (since we probably don't want realization of something lazy on the wrong thread).  If it's List, then is this to just get our own copy?  Perhaps we shoudl type '.Items' as ImmutableArray?  That way the values are realized, and we don't ned to make a copy?